src/manager.js

```diff
@@ -1,72 +1,132 @@
+import { requireLoader } from "@jupyter-widgets/html-manager";
+
 import * as pWidget from "@lumino/widgets";
 
-import { HTMLManager } from "@jupyter-widgets/html-manager";
+import {
+  RenderMimeRegistry,
+  standardRendererFactories,
+} from "@jupyterlab/rendermime";
 
-import * as outputWidgets from "./output";
+import {
+  WidgetManager as JupyterLabManager,
+  WidgetRenderer,
+  output,
+} from "@jupyter-widgets/jupyterlab-manager";
 
-import { ShimmedComm } from "./services-shim";
+import * as base from "@jupyter-widgets/base";
 
-export class ThebeManager extends HTMLManager {
-  get onError() {
-    return this._onError;
-  }
+import * as controls from "@jupyter-widgets/controls";
 
-  registerWithKernel(kernel) {
-    if (this._commRegistration) {
-      this._commRegistration.dispose();
-    }
-    this._commRegistration = kernel.registerCommTarget(
-      this.comm_target_name,
-      (comm, message) => this.handle_comm_open(new ShimmedComm(comm), message)
-    );
-    this.kernel = kernel;
+const WIDGET_MIMETYPE = "application/vnd.jupyter.widget-view+json";
+
+export class ThebeManager extends JupyterLabManager {
+  constructor(kernel) {
+    const context = createContext(kernel);
+    let rendermime;
+    const settings = {
+      saveState: false,
+    };
+    super(context, rendermime, settings);
+    rendermime = createRenderMimeRegistry(this);
```

Comment:
Why isn't this called before `super()`?

Reply:
It's because the super() needs a rendermime, but the rendermime function needs `this`. I'll refactor it so it looks more like the modern day voila `manager.ts`, which only adds the rendermime factory after `super()` is called.